osc/osc.go

```diff
@@ -1,26 +1,37 @@
 package osc
 
 import (
+	"context"
 	"encoding/binary"
-	"fmt"
 	"math"
 	"net"
-	"os"
-	"sync"
 	"time"
+
+	"golang.org/x/sync/errgroup"
 )
 
+type Debugger interface {
+	Debug(msg string, args ...any)
+}
+
 type OSC struct {
 	Destination *net.UDPAddr
 	Client      *net.UDPAddr
 	Conn        *net.UDPConn
+	Debugger    Debugger
+}
+
+func (osc *OSC) Debug(msg string, args ...any) {
+	if osc.Debugger != nil {
+		osc.Debugger.Debug(msg, args...)
+	}
 }
```

Comment:
Does this Debug() method recursively call itself? If the Debugger property is not nil, then call the method Debug().  If the Debugger property is not nil, then call the method Debug().

Reply:
interfaces can be nil. The method of Debug lives on the interface having another method on the Parent struct helps to keep things DRY when trying to write to the debugger.